File: admin/src/api/articles.ts
import api from './index'

export interface Article {
  id: number
  title: string
  content: string
  source_book?: string
  source_chapter?: string
  is_excerpt: boolean
  word_count: number
  reading_time: number
  article_difficulty: number
  status: string
  is_ai_generated: boolean
  created_at: string
  updated_at: string
  tags: Array<{ id: number; name: string; category: string }>
  question_count: number
}

export interface ArticleListParams {
  page?: number
  page_size?: number
  status?: string
  keyword?: string
}

export interface ArticleListResponse {
  items: Article[]
  total: number
  page: number
  page_size: number
}

export interface ArticleCreateParams {
  title: string
  content: string
  source_book?: string
  source_chapter?: string
  is_excerpt?: boolean
  article_difficulty: number
  tag_ids: number[]
}

export const getArticleList = (params: ArticleListParams): Promise<ArticleListResponse> => {
  return api.get('/admin/articles/', { params })
}

export const getArticle = (id: number): Promise<Article> => {
  return api.get(`/admin/articles/${id}`)
}

export const createArticle = (data: ArticleCreateParams): Promise<Article> => {
  return api.post('/admin/articles/', data)
}

export const updateArticle = (id: number, data: Partial<ArticleCreateParams>): Promise<Article> => {
  return api.put(`/admin/articles/${id}`, data)
}

export const deleteArticle = (id: number): Promise<void> => {
  return api.delete(`/admin/articles/${id}`)
}

export const publishArticle = (id: number): Promise<void> => {
  return api.post(`/admin/articles/${id}/publish`)
}

export const archiveArticle = (id: number): Promise<void> => {
  return api.post(`/admin/articles/${id}/archive`)
}

File: backend/app/api/v1/admin/articles.py
from fastapi import APIRouter, Depends, Query, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession
from typing import Optional

from app.database import get_db
from app.api.deps import get_admin_user
from app.schemas.common import ResponseModel
from app.schemas.admin.article import (
    ArticleCreateRequest,
    ArticleUpdateRequest,
    ArticleAdminResponse,
    ArticleListResponseAdmin
)
from app.services.admin.article_service import admin_article_service

router = APIRouter()


@router.get("/", response_model=ResponseModel[ArticleListResponseAdmin])
async def get_article_list(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    status: Optional[str] = Query(None),
    keyword: Optional[str] = Query(None),
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    items, total = await admin_article_service.get_article_list(
        db, page, page_size, status, keyword
    )
    
    return ResponseModel(data=ArticleListResponseAdmin(
        items=items,
        total=total,
        page=page,
        page_size=page_size
    ))


@router.post("/", response_model=ResponseModel[ArticleAdminResponse])
async def create_article(
    data: ArticleCreateRequest,
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    result = await admin_article_service.create_article(db, data)
    return ResponseModel(data=result)


@router.get("/{article_id}", response_model=ResponseModel[ArticleAdminResponse])
async def get_article_detail(
    article_id: int,
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    result = await admin_article_service.get_article_detail(db, article_id)
    if not result:
        raise HTTPException(status_code=404, detail="文章不存在")
    return ResponseModel(data=result)


@router.put("/{article_id}", response_model=ResponseModel[ArticleAdminResponse])
async def update_article(
    article_id: int,
    data: ArticleUpdateRequest,
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    result = await admin_article_service.update_article(db, article_id, data)
    if not result:
        raise HTTPException(status_code=404, detail="文章不存在")
    return ResponseModel(data=result)


@router.delete("/{article_id}", response_model=ResponseModel)
async def delete_article(
    article_id: int,
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    success = await admin_article_service.delete_article(db, article_id)
    if not success:
        raise HTTPException(status_code=404, detail="文章不存在")
    return ResponseModel(message="删除成功")


@router.post("/{article_id}/publish", response_model=ResponseModel)
async def publish_article(
    article_id: int,
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    success = await admin_article_service.publish_article(db, article_id)
    if not success:
        raise HTTPException(status_code=404, detail="文章不存在")
    return ResponseModel(message="发布成功")


@router.post("/{article_id}/archive", response_model=ResponseModel)
async def archive_article(
    article_id: int,
    db: AsyncSession = Depends(get_db),
    admin: dict = Depends(get_admin_user)
):
    success = await admin_article_service.archive_article(db, article_id)
    if not success:
        raise HTTPException(status_code=404, detail="文章不存在")
    return ResponseModel(message="归档成功")
